src/coreclr/vm/method.hpp

```diff
@@ -62,63 +62,76 @@ EXTERN_C VOID STDCALL PInvokeImportThunk();
 
 enum class AsyncMethodKind
 {
-    // Regular methods not returning tasks
-    // These are "normal" methods that do not get other variants.
-    // Note: Generic T-returning methods are NotAsync, even if T could be a Task.
-    NotAsync,
-
-    // Regular methods that return Task/ValueTask
-    // Such method has its actual IL body and there also a synthetic variant that is an
-    // Async-callable thunk. (AsyncVariantThunk)
-    TaskReturning,
-
-    // Task-returning methods marked as MethodImpl::Async in metadata.
-    // Such method has a body that is a thunk that forwards to an Async implementation variant
-    // which owns the original IL. (AsyncVariantImpl)
-    RuntimeAsync,
+    // Method body is emitted as a state machine and uses Async call convention.
+    Async                  = 1,
+    // Not Async method that returns a promise type {Task, Task<T>, ValueTask, ValueTask<T>}
+    ReturnsTaskOrValueTask = 2,
+    // Method has two variants and this is one of them.
```

Comment:
```suggestion
    // MethodDesc represents async calling convention entrypoint of a Task-returning method.
```
Can the description of AsyncVariant be more precise?